deps/rabbit/src/rabbit_amqp_session.erl

```diff
@@ -2719,8 +2719,12 @@ ensure_source(#'v1_0.source'{
         _ ->
             exit_not_implemented("Dynamic source not supported: ~tp", [Source0])
     end;
+ensure_source(undefined, _, _, _, _, _, _, _, _) ->
+    {error, source_address_required};
 ensure_source(Source = #'v1_0.source'{dynamic = true}, _, _, _, _, _, _, _, _) ->
     exit_not_implemented("Dynamic source not supported: ~tp", [Source]);
+ensure_source(#'v1_0.source'{address = undefined, dynamic = false}, _, _, _, _, _, _, _, _) ->
+    {error, source_address_required};
```

Comment:
This function clause is unnecessary.
If the source address is undefined, the code will already return `{error, {bad_address, undefined}}` a few lines further down in https://github.com/rabbitmq/rabbitmq-server/blob/4ba98d8c6c65b357d0f1721b5aabc80b9560b88f/deps/rabbit/src/rabbit_amqp_session.erl#L2756